File: Assets/RFramework/Scripts/Module/BaseModule.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace RFramework
{
    public abstract class BaseModule
    {
        public bool IsComplete;         // 是否已初始化完成
        protected bool m_isUpdate;      // 是否开启Update
        public bool IsUpdate => m_isUpdate;
        
        public abstract void OnStart();
        public abstract void OnUpdate();
        public abstract void OnFinishStart();
        public abstract void OnDestroy();
    }
    
    public interface LateUpdateModuleInterface
    {
        void OnLateUpdate();
    }
}

File: Assets/RFramework/Game/LuaScript/Game/Module/BaseModule.lua
BaseModule = BaseClass()

function BaseModule:OnStart()
	self.IsComplete = true -- 标记初始化完成
	self.IsUpdate = false -- 是否需要Update
end

function BaseModule:OnFinishStart() end
function BaseModule:OnDestroy() end

BaseModule.OnUpdate = nil
BaseModule.OnLateUpdate = nil